### PR TITLE
CORE-16237 Verification rpc processor

### PR DESCRIPTION
**Pull request by emilybowe**
Integration of a HTTP RPC server into the Verification processor.

### CODE REVIEW

File: components/ledger/ledger-verification/src/main/kotlin/net/corda/ledger/verification/LedgerVerificationComponent.kt [hunk 1]
```diff
@@ -36,20 +37,28 @@ class LedgerVerificationComponent @Activate constructor(
...
+    private val subscriptionFactory: SubscriptionFactory
 ) : Lifecycle {
     private var configHandle: Resource? = null
     private var verificationProcessorSubscription: Subscription<String, TransactionVerificationRequest>? = null
 
     companion object {
         private val logger = LoggerFactory.getLogger(this::class.java.enclosingClass)
+        internal const val GROUP_NAME = "verification.ledger.processor"
+        const val CONFIG_HANDLE = "CONFIG_HANDLE"
+        const val SUBSCRIPTION_NAME = "Verification"
```
Comment (mbrkic-r3) [inline components/ledger/ledger-verification/src/main/kotlin/net/corda/ledger/verification/LedgerVerificationComponent.kt]: Constants that are not used here but only in other classes could be moved to those classes

File: components/ledger/ledger-verification/src/main/kotlin/net/corda/ledger/verification/processor/impl/VerificationRpcRequestProcessor.kt [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        else -> {
+            log.warn(errorMessage(externalEventContext, ExternalEventResponseErrorType.TRANSIENT), exception)
+            responseFactory.transientError(externalEventContext, exception)
+        }
+    }
+
+    private fun errorMessage(
+        externalEventContext: ExternalEventContext,
+        errorType: ExternalEventResponseErrorType
+    ) = "Exception occurred (type=$errorType) for flow-worker request ${externalEventContext.requestId}"
```
Comment (mbrkic-r3) [inline components/ledger/ledger-verification/src/main/kotlin/net/corda/ledger/verification/processor/impl/VerificationRpcRequestProcessor.kt]: I guess we should change this to "... verification-worker ..."

File: components/ledger/ledger-verification/src/main/kotlin/net/corda/ledger/verification/LedgerVerificationComponent.kt [hunk 3]
```diff
@@ -74,6 +83,7 @@ class LedgerVerificationComponent @Activate constructor(
                 val newVerificationProcessorSubscription = verificationRequestSubscriptionFactory.create(
                     event.config.getConfig(MESSAGING_CONFIG)
                 )
+                initialiseRpcSubscription()
```
Comment (mbrkic-r3) [inline components/ledger/ledger-verification/src/main/kotlin/net/corda/ledger/verification/LedgerVerificationComponent.kt]: I think this should go below `if (event.status == LifecycleStatus.UP) {`, similar as it was for Uniqueness.

File: components/ledger/ledger-verification/src/main/kotlin/net/corda/ledger/verification/LedgerVerificationComponent.kt [hunk 4]
```diff
@@ -36,20 +36,21 @@ class LedgerVerificationComponent @Activate constructor(
...
     @Reference(service = VerificationSubscriptionFactory::class)
-    private val verificationRequestSubscriptionFactory: VerificationSubscriptionFactory
+    private val verificationRequestSubscriptionFactory: VerificationSubscriptionFactory,
 ) : Lifecycle {
     private var configHandle: Resource? = null
     private var verificationProcessorSubscription: Subscription<String, TransactionVerificationRequest>? = null
 
     companion object {
         private val logger = LoggerFactory.getLogger(this::class.java.enclosingClass)
+        const val RPC_SUBSCRIPTION = "RPC_SUBSCRIPTION"
```
Comment (mbrkic-r3) [inline components/ledger/ledger-verification/src/main/kotlin/net/corda/ledger/verification/LedgerVerificationComponent.kt:46]: Could you please check whether this name needs to be unique (could it cause issues for CombinedWorker where different Processors register different objects using the same name)?

**Review by mbrkic-r3 — APPROVED** [review]
LGTM

**Review by mcgovc — APPROVED** [review]
LGTM - build file changes

**Review by blsemo — APPROVED** [review]
Looks fine to me.

**Pull request merged**